packages/core/src/odata/common/request-builder/batch/batch-response-parser.ts

```diff
@@ -0,0 +1,376 @@
+  const parts = response.split(`--${boundary}`).map(part => part.trim());
+
+  if (parts.length >= 3) {
+    return parts.slice(1, parts.length - 1);
+  }
+
+  throw new Error(
+    'Could not parse batch response body. Expected at least two response boundaries.'
+  );
+}
+
+/**
+ * Parse the entity name from the metadata uri. This should be the `__metadata` property of a single entity in the response.
+ * @param uri The URI to parse the entity name from
+ * @returns The entity name.
+ */
+export function parseEntityNameFromMetadataUri(
+  uri: string | undefined
+): string {
+  if (!uri) {
+    throw new Error(
+      `Could not retrieve entity name from metadata. URI was: '${uri}'.`
+    );
+  }
+  const [pathBeforeQuery] = uri.split('?');
+  const [pathBeforeKeys] = pathBeforeQuery.split('(');
+  const uriParts = pathBeforeKeys.split('/');
+
+  // Remove another part in case of a trailing slash
+  const entityName = uriParts.pop() || uriParts.pop();
+  if (!entityName) {
+    throw Error(
+      `Could not retrieve entity name from metadata. Unknown URI format. URI was: '${uri}'.`
+    );
+  }
+  return entityName;
+}
+
+/**
+ * Retrieve the constructor for a specific single response body.
+ * @param responseBody The body of a single response as an object.
+ * @param entityToConstructorMap Mapping between entity names and their respective constructors.
+ * @param responseDataAccessor Response data access module.
+ * @returns The constructor if found in the mapping, undefined otherwise.
+ */
+export function getConstructor(
+  responseBody: Record<string, any>,
+  entityToConstructorMap: Record<string, Constructable<EntityBase>>,
+  responseDataAccessor: ResponseDataAccessor
+): Constructable<EntityBase> | undefined {
+  const entityJson = responseDataAccessor.isCollectionResult(responseBody)
+    ? responseDataAccessor.getCollectionResult(responseBody)[0]
+    : responseDataAccessor.getSingleResult(responseBody);
+
+  const entityUri = entityJson?.__metadata?.uri;
+  if (entityUri) {
+    return entityToConstructorMap[parseEntityNameFromMetadataUri(entityUri)];
+  }
+
+  logger.warn('Could not parse constructor from response body.');
+}
+
+/**
+ * Parse the HTTP code of response.
+ * @param response String representation of the response.
+ * @returns The HTTP code.
+ */
+export function parseHttpCode(response: string): number {
+  const group = response.match(/HTTP\/\d\.\d (\d{3}).*?/);
+
+  if (group) {
+    return parseInt(group[1].toString());
+  }
+
+  throw new Error('Cannot parse http code of the response.');
+}
+
+/**
+ * Create a function to transform the parsed response body to a list of entities of the given type or an error.
+ * @param body The parsed JSON reponse body.
+ * @param responseDataAccessor Response data access module.
+ * @param deserializer Entity deserializer.
+ * @returns A function to be used for transformation of the read response.
+ */
+function asReadResponse(
```

Comment:
return type in method signature?

Reply:
I think this is not necessary because this is returning a function and the function within has a signature which is used transitively for this one. Adding it would make it harder to read. I checked that the type is inferred correctly and that the documentation also contains the correct type, which both do. Is that ok for you?